### ISSUE TITLE
generate_graph_parents: tasks appearing at the wrong cycle point

### ISSUE THREAD

**Issue by wxtim** [cylc/cylc-flow#5490]
Reported by two users.

### Description

Under some circumstances tasks which should only be carried out once, at the initial cycle point appear in the GUI at Cycle point other than the ICP.

In both cases the following pattern of dependencies has been seen - 

```
R1 = run_once_task
TX = run_once_task[-PTX] => task_for_this_cycle
```

### Reproducible Example

```ini
[scheduler]
    allow implicit tasks = True
[scheduling]
    initial cycle point = 2023
    final cycle point = +PT9H
    [[graph]]
        R1 = run_once_at_midnight
        T00 = run_once_at_midnight[-PT0H] => every_cycle
        T03 = run_once_at_midnight[-PT3H] => every_cycle
        T06 = run_once_at_midnight[-PT6H] => every_cycle
[runtime]
    [[every_cycle]]
        script = sleep 60  # Added to stop the bug being transitory.

```

In the GUI we get the following 

![image](https://user-images.githubusercontent.com/26465611/234325742-c80b6a6f-293c-462d-9555-b55df5ac7b11.png)

(and the equivalent wrong dependency in the graph view)

### Expected Behaviour

The graph should look thus:

![image](https://user-images.githubusercontent.com/26465611/234326393-24d98260-de44-4680-9525-ad604d84f53d.png)

### Additional information

In the example above looking at the database and log files strongly suggests that the workflow is behaving as intended:

```sqlite
sqlite> SELECT cycle, name FROM task_jobs;
20230101T0000Z|every_cycle
20230101T0000Z|run_once_at_midnight
20230101T0300Z|every_cycle
20230101T0600Z|every_cycle
```

```bash
> ls ~/cylc-run/example/runN/log/job/20230101T0*/
~/cylc-run/example/runN/log/job/20230101T0000Z/:
every_cycle  run_once_at_midnight
~/cylc-run/example/runN/log/job/20230101T0300Z/:
every_cycle
~/cylc-run/example/runN/log/job/20230101T0600Z/:
every_cycle
```

**Comment by oliver-sanders** [comment]
(changed the title to avoid confusion with "ghost tasks" which have a special meaning in the data store)

**Comment by oliver-sanders** [comment]
Have managed to track this down to `cylc.flow.taskdef.generate_graph_parents` which is feeding the incorrect cycle point into the data store.

**Comment by hjoliver** [comment]
Fortunately that method is only used by the datastore, so this bug won't affect scheduling at all.

**Comment by hjoliver** [comment]
Got it ...

**Comment by oliver-sanders** [comment]
Closed by #5495
